src/modules/ens/sagas.ts

```diff
@@ -365,7 +365,10 @@ export function* ensSaga(builderClient: BuilderClient, ensApi: ENSApi, worldsAPI
         REGISTRAR_ADDRESS,
         signer
       )
-      let domains: string[] = yield call([marketplace, 'fetchENSList'], address)
+      const defaultPageSize = 12
+      const { first = defaultPageSize, skip = 0 } = action.payload
+      let domains: string[] = yield call([marketplace, 'fetchENSList'], address, first, skip)
+      const totalDomains: number = yield call([marketplace, 'fetchENSListCount'], address)
       const bannedDomains: string[] = yield call(fetchBannedDomains)
```

Comment:
These 3 requests can be done concurrently to speed up the fetch process, WDYT?

Reply:
totally! good call!